### PR TITLE
Adds left menu search component

### LINKED ISSUE #1362: New design todo list
## Yet to be done

- [x] bug: if opening a tab on the parameter tab https://dev.algorea.org/en/a/5948378700861088397;p=39530140456452546;a=0/parameters, then switching to the "content" tab makes an empty tab (moved to #1389)
- [x] mini-bug: https://dev.algorea.org/branch/feature/left-menu-search/en/a/1646068694606368976;p=4702,1352246428241737349,1070539248445908636,61555878157023179;a=0 -> in the left menu, the back button is squeezed  (moved to #1390)
- [ ] breadcrumb: there should be a fixed space on the left so that there is no "jump" when the score is known.
- [ ] "description" in items:    font-weight 300 + Line-height 1.5rem,   2 rem margin down
- [ ] left menu: save the space use for the score ring so that we do not have a line jump when the score appear (for instance [here](https://beta.opentezos.com/en/a/2446909545349166564;p=1,1484719710020206758;pa=0) and you click on "Introduction to other consensus algorithms"
- [ ] increase the scroll threshold to switch the top bar from breadcrumb to "title only"... so that we cannot have twice the prev/next/top navigation displayed
- [ ] center vertically all loaders (menu and content)
- [ ] left menu: no empty space on top of the menu when there is no activities/skills/groups selector
- [ ] There are twice (?) the margin between the error message and the title ? https://dev.algorea.org/branch/new-design/item-loading/en/a/1980584647557587953;p=4702,4102;pa=0
- [ ] Would it be possible to animated the line under the tab when changing tabs?
- [ ] lighter "starting activity": https://www.sketch.com/s/0a19c027-9893-420d-90fb-6694a72faccf/a/r9qxnOD
- [ ] children list (and other places): there is a padding under the list which adds to the 64px page padding... it should be 64px all included
- [ ] children list: display title in dark-text-color
- [x] group pages 
- [ ] [forum](https://www.sketch.com/s/0a19c027-9893-420d-90fb-6694a72faccf/a/3ooaZ8w)
- [x] search in left menu (the design already)
- [ ] [observation mode](https://www.sketch.com/s/0a19c027-9893-420d-90fb-6694a72faccf/a/aMMjvvq)
- [ ] [forms](https://www.sketch.com/s/0a19c027-9893-420d-90fb-6694a72faccf/a/OeekZZR)

## On hold
- [ ] bug with the scroll (old bug which was fixed): the vertical scrolling on tasks where there is scrolling in the iframe of the task ([for instance here](https://beta.opentezos.com/en/a/2446909545349166564;p=1,1484719710020206758;pa=0), [on the dev content](https://dev.algorea.org/en/a/5515805070397494516;p=7528142386663912287,7523720120450464843,6963488196048051083;a=0)) is sometimes "caught" by the app and not by iframe... (seems to appear only when the left menu is being displayed (?)) 
https://github.com/France-ioi/AlgoreaFrontend/assets/1053150/746a118f-76ff-487a-ba1c-bc151625081f => **on hold, reported to Michel**

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #1362

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

Search input

## Test cases

- [ ] Case 1:
  1. Given I am the temp user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/left-menu-search/en/a/4102;p=4702;pa=0/progress/chapter-user-progress)
  3. Then I see search input in left menu

### CODE REVIEW

**Comment by smadbe** [comment]
Please add a feature flag to allow disabling it from the config.

And the search result pane is not implemented in this PR?

**Comment by smadbe** [comment]
I have plugged the actual call to the search into the UI. @Iloveall could you finish the work, so basically add a design into `left-search-result.component`. I was unable to reuse the left-nav-tree as it was too different, but basically the design should be quite similar.
- probably you can use the `ph-folders` icon for chapters as they do not really open on click). 
- the close search icon should be similar to the "back to parent" line but with a "x" instead.

**Comment by smadbe** [comment]
I assume just the "X" in the light grey circle would be enough, no need for 2 circles 
Screenshot 2023-06-14 at 22 11 01

**Comment by smadbe** [comment]
Maybe you can do as [in the design](https://www.sketch.com/s/0a19c027-9893-420d-90fb-6694a72faccf/a/x0rwJx0): the left arrow and the number of search results "x activities found" (or "activity" if 1)

File: src/app/core/components/left-search-result/left-search-result.component.html [hunk 1]
```diff
@@ -1,17 +1,48 @@
...
+    <button
+      class="alg-button-icon back-button rounded"
+      icon="ph ph-arrow-left"
+      pButton
+    ></button>
+    <p class="back-button-title" i18n>
+      {{ data.length }}
+      {{ data.length | i18nPlural: {
+        '=1': 'activity found',
+        'other': 'activities found'
```
Comment (smadbe) [inline src/app/core/components/left-search-result/left-search-result.component.html:12]: In French, the plural is for >1 (so 0 is singular). It should be supported then (having 3 cases if i18n does not know the per-language differences)

**Review by smadbe — APPROVED** [review]
After these 3 last remarks, it should be good.

File: src/app/core/components/left-search-result/left-search-result.component.html [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
+    </p>
+  </div>
+</div>
 
-<p *ngIf="data === undefined">
-  Type at least 3 char to search content
-</p>
+<div class="container" *ngIf="data === undefined">
+  <alg-message-info icon="ph-duotone ph-warning-circle" i18n>
+    Type at least 3 char to search content
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/core/components/left-search-result/left-search-result.component.html]: Fixed

File: src/app/core/components/left-search-result/left-search-result.component.html [hunk 3]
```diff
@@ -1,23 +1,17 @@
...
-        '=1': 'activity found',
-        'other': 'activities found'
-      } }}
-    </p>
-  </div>
+<div class="container" *ngIf="data !== undefined">
+  <alg-left-menu-back-button (close)="close.emit()">
+    {{ data.length }}
+    {{ data.length | i18nPlural: {
+      '=0': 'activity found',
```
Comment (smadbe) [inline src/app/core/components/left-search-result/left-search-result.component.html:5]: In English, it is  '0 activities found'

Comment (Iloveall) [reply src/app/core/components/left-search-result/left-search-result.component.html]: Fixed

**Pull request merged**